File: final_qualifying_work/Projects/server/Models/Dtos/GPSDataDto.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace server.Models.Dtos
{
    public class GPSDataDto
    {
        public ulong RecId { get; set; }
        public DateTime RecDatetime { get; set; }
        public ulong TripId { get; set; }
        public float LatitudeDEG { get; set; }
        public float LongitudeDEG { get; set; }
        public float? AccuracyM { get; set; }
        public float? SpeedKMH { get; set; }
        public ushort? BearingDEG { get; set; }
    }
}

File: final_qualifying_work/Projects/data_provider_app/app/src/main/java/com/example/data_provider_app/dto/GPSDataDto.kt
package com.example.data_provider_app.dto

import java.time.LocalDateTime

data class GPSDataDto(
    val recId: ULong,
    val recDatetime: LocalDateTime,
    val tripId: ULong,
    val latitudeDEG: Float,
    val longitudeDEG: Float,
    val accuracyM: Float?,
    val speedKMH: Float?,
    val bearingDEG: UShort?
)
